recipe/meta.yaml

```diff
@@ -671,17 +681,12 @@ outputs:
         - cmake
         - ninja
       host:
-        - {{ pin_subpackage("libarrow-all", exact=True) }}
-        - clangdev {{ llvm_version }}
-        - llvmdev {{ llvm_version }}
-        - zlib
-        - cython
-        - numpy
-        - python
-        - setuptools
-        - setuptools-scm
-      run:
-        # full set of libs because run-exports from libarrow-all aren't picked up
+        # We add all libarrow package dependencies on host in order to build pyarrow
+        # once with all capabilities. We require the exact hash versions but we don't
+        # want those to be installed when running for pyarrow-core as we want a
+        # low memory footprint for the pyarrow-core package.
+        # We cannot use them on build as we want the exact hash version built on the
+        # recipe to be linked, otherwise we get mismatching package errors.
```

Comment:
I don't understand this part of the comment

Reply:
I was trying to explain why they are on the host section and not on the build section. To be honest I added the comment because I tried to add them on the build section instead of host and I was getting a mismatching package error. I am removing the comment because it is not clear at all and I am not sure how to explain it better. Please do let me know if you think I should add something about that.